### ISSUE TITLE
Serious bug in identifyOfftargetSites.py causes bona fide off-targets to be missed

### ISSUE THREAD

**Issue by staciawyman** [aryeelab/guideseq#51]
There is a very serious bug in the identiyOfftargetSites.py file. The alignSequences routine calls reverseComplement on the target sequence which has been extracted from the reference genome. When the reference genome sequence is lowercase, reverseComplement routine **does not complement** because the translation command just includes uppercase letters. 

The impact of this is that **bona fide off-targets on the antisense strand in lowercase regions of the genome are filtered out** when they should not be.  I found this bug when I ran our samples for the EMX1 guide with hg38 and the OT1 site on Chr5 was missing from the output, but was found if I ran it with hg19. The genomic sequence at OT1 is uppercase in hg19 and lowercase in hg38. 

I fixed this by adding the lowercase letters to the reverse complement routine: 

```
def reverseComplement(sequence):
    transtab = string.maketrans("ACGTacgt", "TGCAtgca")
    return sequence.translate(transtab)[::-1]
```

**Comment by staciawyman** [comment]
Any progress on fixing this bug? Genuine off-targets are being missed because of this.

**Comment by martinaryee** [comment]
Hi Stacia,

We’re testing the fix now and plan to have it rolled out this week.

Thanks again for the report!
Martin

**Comment by staciawyman** [comment]
Terrific! Thanks for getting on this.

**Comment by sowmyaiyer** [comment]
Hi Stacy,
Thank you for bringing this to our attention and for your suggested fix - much appreciated. We made a slight modification to your code suggestion to avoid running into problems downstream in the analysis. Please see below and if you don't see any problems, we can roll this out very quickly.

`def reverseComplement(sequence):
    transtab = string.maketrans("ACGTacgt", "TGCATGCA")
    return sequence.translate(transtab)[::-1]`

**Comment by staciawyman** [comment]
Yes that's great. Also has the benefit of preserving the lowercase letters when returning the sequence for output in the "Sequence" column of the identifiedOfftargets file.

**Comment by martinaryee** [comment]
This is now fixed in v1.0.2: https://github.com/aryeelab/guideseq/releases/tag/v1.0.2

Thanks for reporting!
